### PR TITLE
added adb setup + uad setup

### PR DESCRIPTION
**Pull request by guruswarupa**
## Type of Change
- [x] New feature

## Description
Installs Android tools (adb) and installs Universal Android debloater Next Gen.
- run uad command to execute  Universal Android debloater
## Checklist
- [x] My code adheres to the coding and style guidelines of the project.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have made corresponding changes to the documentation.
- [x] My changes generate no errors/warnings/merge conflicts.

### CODE REVIEW

File: core/tabs/applications-setup/android-debloat.sh [hunk 1]
```diff
@@ -0,0 +1,44 @@
+#!/bin/sh -e
+
+. ../common-script.sh
+
+install_adb() {
+    if ! command_exists adb ; then
+        printf "%b\n" "${YELLOW}Installing ADB..${RC}."
```
Comment (adamperkowski) [inline core/tabs/applications-setup/android-debloat.sh:7]:
```suggestion
        printf "%b\n" "${YELLOW}Installing ADB...${RC}."
```

File: core/tabs/applications-setup/android-debloat.sh [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+                "$ESCALATION_TOOL" "$PACKAGER" install android-sdk-platform-tools
+                ;;
+            zypper)
+                "$ESCALATION_TOOL" "$PACKAGER" install android-tools
+                ;;
+            pacman)
+                "$ESCALATION_TOOL" "$PACKAGER" -S android-tools
+                ;;
+            yum)
+                "$ESCALATION_TOOL" "$PACKAGER" install android-tools
```
Comment (adamperkowski) [inline core/tabs/applications-setup/android-debloat.sh:19]:
```suggestion
                "$ESCALATION_TOOL" "$PACKAGER" install -y android-sdk-platform-tools
                ;;
            zypper)
                "$ESCALATION_TOOL" "$PACKAGER" --non-interactive install android-tools
                ;;
            pacman)
                "$ESCALATION_TOOL" "$PACKAGER" -S --noconfirm android-tools
                ;;
            yum)
                "$ESCALATION_TOOL" "$PACKAGER" install -y android-tools
```

File: core/tabs/applications-setup/android-debloat.sh [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+                ;;
+            *)
+                printf "%b\n" "${RED}The script does not support your Distro. Install manually..${RC}"
```
Comment (adamperkowski) [inline core/tabs/applications-setup/android-debloat.sh:22]:
```suggestion
                printf "%b\n" "${RED}Unsupported package manager: "$PACKAGER"${RC}"
```

File: core/tabs/applications-setup/android-debloat.sh [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+                ;;
+        esac
+    else
+        printf "%b\n" "${GREEN}ADB is already installed.${RC}"
+    fi
+}
+
+install_universal_android_debloater() {
+    if ! command_exists uad; then
+        printf "%b\n" "${YELLOW}Installing Universal Android Debloater..${RC}."
```
Comment (adamperkowski) [inline core/tabs/applications-setup/android-debloat.sh:32]:
```suggestion
        printf "%b\n" "${YELLOW}Installing Universal Android Debloater...${RC}."
```

File: core/tabs/applications-setup/android-debloat.sh [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+        curl -sSLo "${HOME}/uad" "https://github.com/Universal-Debloater-Alliance/universal-android-debloater-next-generation/releases/download/v1.1.0/uad-ng-linux"
+        "$ESCALATION_TOOL" chmod +x "${HOME}/uad"
+        "$ESCALATION_TOOL" mv "${HOME}/uad" /usr/local/bin/uad
+    else
+        printf "%b\n" "${GREEN}Universal Android Debloater is already installed. Run 'uad' command to execute...${RC}"
```
Comment (adamperkowski) [inline core/tabs/applications-setup/android-debloat.sh:37]:
```suggestion
        printf "%b\n" "${GREEN}Universal Android Debloater is already installed. Run 'uad' command to execute.${RC}"
```

**Review by adamperkowski — CHANGES_REQUESTED** [review]

File: core/tabs/applications-setup/android-debloat.sh (hunk 3 above)
Comment (nnyyxxxx) [inline core/tabs/applications-setup/android-debloat.sh:22]:
```suggestion
                printf "%b\n" "${RED}Unsupported package manager: "$PACKAGER"${RC}"
                exit 1
```

Comment (adamperkowski) [reply core/tabs/applications-setup/android-debloat.sh:22]: https://github.com/ChrisTitusTech/linutil/pull/592#discussion_r1769638445

Comment (nnyyxxxx) [reply core/tabs/applications-setup/android-debloat.sh:22]: u forgot exit 1 nub

Comment (adamperkowski) [reply core/tabs/applications-setup/android-debloat.sh:22]: Oh yes. Sorry

**Review by adamperkowski — APPROVED** [review]

File: core/tabs/applications-setup/android-debloat.sh [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+                ;;
+            zypper)
+                "$ESCALATION_TOOL" "$PACKAGER" --non-interactive install android-tools
+                ;;
+            pacman)
+                "$ESCALATION_TOOL" "$PACKAGER" -S --noconfirm android-tools
+                ;;
+            yum)
+                "$ESCALATION_TOOL" "$PACKAGER" install -y android-tools
+                ;;
```
Comment (nnyyxxxx) [inline core/tabs/applications-setup/android-debloat.sh:20]: remove yum

Comment (adamperkowski) [reply core/tabs/applications-setup/android-debloat.sh:20]: replace with dnf

Comment (guruswarupa) [reply core/tabs/applications-setup/android-debloat.sh:20]: In the docs it's given to use yum.

Comment (guruswarupa) [reply core/tabs/applications-setup/android-debloat.sh:20]: > replace with dnf

You sure dnf has that package?

Comment (adamperkowski) [reply core/tabs/applications-setup/android-debloat.sh:20]: > You sure dnf has that package?

https://packages.fedoraproject.org/pkgs/android-tools/android-tools/index.html
From the wiki, dnf uses packages.fedoraproject.org as a resource.

**Review by adamperkowski — APPROVED** [review]

**Review by ChrisTitusTech — APPROVED** [review]

**Pull request merged**